crates/interpreter/src/interpreter/shared_stack.rs

```diff
@@ -0,0 +1,710 @@
+        shared_stack.new_context();
+        assert_eq!(shared_stack.page_idx, 1);
+        assert_eq!(shared_stack.pages.len(), 2);
+        assert_eq!(shared_stack.page().checkpoints.len(), 1);
+        assert_eq!(shared_stack.page().checkpoints[0], 0);
+        assert_eq!(shared_stack.context_len, 0);
+    }
+
+    #[test]
+    fn new_consecutive() {
+        let mut shared_stack = SharedStack::new();
+        for i in 0..3 {
+            for j in 0..7 {
+                shared_stack.new_context();
+                assert_eq!(shared_stack.page_idx, i);
+                assert_eq!(shared_stack.pages.len(), i + 1);
+                assert_eq!(shared_stack.context_len, 0);
+                assert_eq!(shared_stack.page().checkpoints.len(), j + 1);
+                assert_eq!(
+                    shared_stack.page().checkpoints.last().cloned().unwrap(),
+                    shared_stack.buffer().len()
+                );
+
+                let new_len = STACK_LIMIT / 2;
+                unsafe { shared_stack.buffer_mut().set_len(new_len * (j + 1)) };
+                shared_stack.context_len = new_len;
+            }
+        }
+    }
+
+    #[test]
+    fn pop() {
+        let mut shared_stack = SharedStack::new();
+        shared_stack.new_context();
+
+        shared_stack.buffer_mut().push(U256::from(1));
+        assert_eq!(shared_stack.buffer()[0], U256::from(1));
+        shared_stack.context_len += 1;
+        assert_eq!(shared_stack.pop(), Ok(U256::from(1)));
+    }
+
+    #[test]
+    fn pop_underflow() {
+        let mut shared_stack = SharedStack::new();
+        shared_stack.new_context();
+
+        assert_eq!(shared_stack.pop(), Err(InstructionResult::StackUnderflow));
+
+        // pop underflow in a new empty context
+        unsafe { shared_stack.page_mut().buffer.set_len(STACK_LIMIT / 2) }
+        shared_stack.new_context();
+        assert_eq!(shared_stack.pop(), Err(InstructionResult::StackUnderflow));
+    }
+
+    #[test]
+    fn push() {
+        let mut shared_stack = SharedStack::new();
+        shared_stack.new_context();
+
+        let one = U256::from(1);
+        let res = shared_stack.push(one);
+        assert_eq!(res, Ok(()));
+        assert_eq!(shared_stack.page_mut().buffer[0], one);
+
+        let two = U256::from(2);
+        let res = shared_stack.push(two);
+        assert_eq!(res, Ok(()));
+        assert_eq!(shared_stack.page_mut().buffer[1], two);
+    }
+
+    #[test]
+    fn push_stack_overflow() {
+        let mut shared_stack = SharedStack::new();
+        shared_stack.new_context();
+
+        for _ in 1..=STACK_LIMIT {
+            assert_eq!(shared_stack.push(U256::ZERO), Ok(()));
+        }
+
+        assert_eq!(shared_stack.page_mut().buffer.len(), STACK_LIMIT);
+        assert_eq!(shared_stack.len(), STACK_LIMIT);
+        assert_eq!(
+            shared_stack.push(U256::ZERO),
+            Err(InstructionResult::StackOverflow)
+        );
+    }
+
+    #[test]
+    fn push_slice() {
```

Comment:
Please keep the test cases from https://github.com/bluealloy/revm/pull/837
There's multiple things to test for this function

Reply:
The thing is that this branch is not merged with the `main`, so it has old changes. This should be fixed after the git merge but it is good to be aware of it.

Reply:
Ah sorry, i didnt see that